src/vos/vos_tree.c

```diff
@@ -612,7 +625,35 @@ svt_rec_free(struct btr_instance *tins, struct btr_record *rec, void *args)
 		vos_bio_addr_free(pool, addr, irec->ir_size);
 	}
 
-	return umem_free(&tins->ti_umm, rec->rec_off);
+	if (!overwrite)
+		return umem_free(&tins->ti_umm, rec->rec_off);
+
+	/** Find an empty slot for the deferred free */
+	for (i = dth->dth_rsrvd_cnt - 1; i >= 0; i--) {
```

Comment:
I'm not overwriting anything here.  I allocate enough space in each one to be able to do a deferred free for everything that is published.  You can only do this once per reserved value.

Reply:
I mean that you need to find out the former reserved "act" that belong to this EV record, then defer_free(). But your current logic only find out one "act", but the "act" may be for other EV record.

Reply:
Yeah, it needs. a bit of work.   Can I leave this to future patch?

I think the right approach is to have a separate rsrvds list.

It should be a corner case where it will fail.   I allocate double space for each entry in rsrvd_scm and by searching in reverse, it makes it somewhat unlikely to hit #2 though not impossible.   I'd like to get this patch landed and fix this in a future patch

Reply:
It is OK for me to resolve the issues in another patch as long as it can pass the CI tests.

Two points to be improved:

1. Prepare enough slots for reservation and defer_free.

2. Guarantee that the slot for defer_free is after related reservation slot. One possible solution is that we can prepare two arrays (with the sizeof of dth_modification_cnt for each): one if for reservation, the other is defer_free(). When publish, handle the reservation array firstly, then there will not be order related issue.